Expands sandbox setup documentation (#552)
edx/edx-proctoring: docs/developing.rst

How do I use proctoring on devstack?

> ------------------------------------

Author: By way of explanation, we skipped a section heading level I found I wanted to use

        * Go to ``/admin/verify_student/manualverification/`` on your sandbox
        * Create a record for the given user, with status "approved".

Backend-specific Information
^^^^^^^^^^^^^^^^^^^^^^^^^^^^

One of the main motivations for setting up a sandbox to test
proctoring is having an externally accessible system which can be
accessed by our proctoring providers' systems. This enables more
thorough end-to-end testing.

> To enable proctoring in a way that won't be overridden by ansible

Reviewer: After one sshing to the running sandbox and updating the yaml file, is any action required after that change?  Do you need to rerun ansible on the sandbox (e.g. as describe in the `Updating code` section of [this](https://openedx.atlassian.net/wiki/spaces/EdxOps/pages/13960183/Sandboxes) document)? Or does that occur as part of restarting the lms and cms processes?

Author: @MichaelRoytman do you think the update I put for this is enough?

Reviewer: I think so! The only thing I notice is that the bullet points under `In addition to adding these configurations, you'll also need to set up
a user which PT can authenticate as.` aren't displaying correctly when viewing the file in GitHub.

Author: Good eye, thanks!

plays, you can add the following to a sandbox's
``/edx/app/edx_ansible/server-vars.yml`` at the end of the
``EDXAPP_FEATURES`` array::

  EDX_APP_FEATURES:
    MILESTONES_APP: true
    ...
    ENABLE_API_DOCS: true
    ENABLE_SPECIAL_EXAMS: true

  PROCTORING_SETTINGS:
    MUST_BE_VERIFIED_TRACK: False

  EDXAPP_PROCTORING_BACKENDS:
    ...

Placing these configurations here (rather than the more generic
locations mentioned in `the README`_) will allow us to leverage the
power of the ansible playbooks used to construct and administer
sandboxes.

The contents of ``EDXAPP_PROCTORING_BACKENDS`` will depend on which

> backend you're interested in testing. It's nececssary to provide a

Reviewer: ```suggestion
backends you're interested in testing. It's necessary to provide a
```
Re: `backend` versus `backends`, the way this is phrased implies, to me, that the sandbox can be configured with only one backend. I think making backends plural clarifies that you can configure your sandbox with more than one backend, which may be a helpful distinction for people new to proctoring.

``DEFAULT`` backend.

Proctortrack
""""""""""""

As will be the case with all REST backends implementing `our spec`_, one
doesn't need to configure much to get Proctortrack working on a
sandbox, e.g.::
  EDXAPP_PROCTORING_BACKENDS:
    DEFAULT: 'proctortrack'
    proctortrack:

>       client_id: "<you'll need to fill this these in wth credentials from Proctortrack>"

Reviewer: ```suggestion
      client_id: "<you'll need to fill this in with credentials from Proctortrack>"
```

>       client_secret: "<you'll need to fill this these in wth credentials from Proctortrack>"

Reviewer: ```suggestion
      client_secret: "<you'll need to fill this in with credentials from Proctortrack>"
```
